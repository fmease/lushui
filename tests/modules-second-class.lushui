;; TEST fail check

RECORD: Type = Type

r0: RECORD = am-i-a-record

module am-i-a-record: =
    module am-i-a-record-too: =

r1: RECORD = am-i-a-record.am-i-a-record-too

use am-i-a-record.am-i-a-record-too

r2: ANOTHER-RECORD = am-i-a-record-too
