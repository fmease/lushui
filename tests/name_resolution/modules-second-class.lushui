;; TEST fail check

RECORD: Type = Type

r0: RECORD = am-i-a-record

module am-i-a-record of
    @public
    module am-i-a-record-too of

r1: RECORD = am-i-a-record.am-i-a-record-too

use am-i-a-record.am-i-a-record-too

r2: ANOTHER-RECORD = am-i-a-record-too
