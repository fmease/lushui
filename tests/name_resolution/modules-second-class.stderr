error[E023]: module used as if it was a value
 >>> ${DIRECTORY}/name_resolution/modules-second-class.lushui:5:14
  |
5 | r0: RECORD = am-i-a-record
  |              ^^^^^^^^^^^^^ 
  |
 help: modules are not first-class citizens, consider utilizing records for such cases instead

error[E023]: module used as if it was a value
  >>> ${DIRECTORY}/name_resolution/modules-second-class.lushui:10:28
   |
10 | r1: RECORD = am-i-a-record.am-i-a-record-too
   |                            ^^^^^^^^^^^^^^^^^ 
   |
  help: modules are not first-class citizens, consider utilizing records for such cases instead

error[E021]: binding `ANOTHER-RECORD` is not defined in this scope
  >>> ${DIRECTORY}/name_resolution/modules-second-class.lushui:14:5
   |
14 | r2: ANOTHER-RECORD = am-i-a-record-too
   |     ^^^^^^^^^^^^^^ 
   |

error[E023]: module used as if it was a value
  >>> ${DIRECTORY}/name_resolution/modules-second-class.lushui:14:22
   |
14 | r2: ANOTHER-RECORD = am-i-a-record-too
   |                      ^^^^^^^^^^^^^^^^^ 
   |
  help: modules are not first-class citizens, consider utilizing records for such cases instead

error: aborting due to 4 previous errors

